price_calculator.rb

```diff
@@ -0,0 +1,9 @@
+module Price_calculator 
+  def total_amount 
+    self.items.map{ |i| i.unit_price * i.quantity }.inject(:+)
+  end
+
+  def saved_amount 
+    total_amount - self.items.map{ |i| i.price }.inject(:+)
```

Comment:
Used another method that does the same thing. Like self.items.sum { |i| i.price }